hooks/pre/31-upgrade_storage_engine.rb

```diff
@@ -0,0 +1,56 @@
+MONGO_ENGINE_COMPLETE = '/etc/foreman-installer/.mongo_engine_upgrade'.freeze
+MONGO_DIR = "/var/lib/mongodb"
+
+# check diskspace
+def disk_space
+  logger.info 'Checking available diskspace in /tmp for upgrade'
+  total_space = `df -k --total --exclude-type=tmpfs`.split("\n")[-1].split[3].to_i
+  mongo_size = File.directory?(@MONGO_DIR) ? `du -s  /var/lib/mongodb/`.split[0].to_i : 0
+  if total_space < mongo_size
+    logger.error 'The free space is less than the size of the Mongo db directory, please add additional space to /tmp and try again, exiting.'
+    kafo.class.exit(1)
+  else
+    logger.info 'Diskspace looks good, continuing to upgrade'
+    migration
+  end
+end
+
+def migration
+  export_dir = '/tmp/engine_upgrade'
+  logger.info 'Stopping services except Mongo'
+  Kafo::Helpers.execute('katello-service stop --exclude rh-mongodb34-mongod')
+  logger.info "Starting mongodump to #{export_dir}"
+  Kafo::Helpers.execute("mkdir -p #{export_dir}/backup")
+  Kafo::Helpers.execute("mongodump --host localhost --out #{export_dir}")
+  logger.info 'Export done, stopping Mongo to remove old datafiles'
+  logger.info "Moving contents from #{@MONGO_DIR}"
+  Kafo::Helpers.execute("mv #{@MONGO_DIR}/* #{export_dir}/backup")
+  logger.info 'Turning on mongo and starting import'
+  Kafo::Helpers.execute('katello-service start --only rh-mongodb34-mongod')
+  Kafo::Helpers.execute("mongod --storageEngine wiredTiger --dbpath #{@MONGO_DIR}")
+  Kafo::Helpers.execute("mongorestore --host localhost --db pulp_database --quiet --dir #{@MONGO_DIR}")
+  logger.info "Import done, removing old data in #{export_dir}/backup"
+  Kafo::Helpers.execute("rm -rf #{export_dir}/backup")
+  # update Hiera to wiredTiger for installer run
+  logger.info 'Changing custom Hiera to use wiredTiger for installer Puppet run.'
+  Kafo::Helpers.execute("sed -i -e 's/run the installer with --upgrade-mongo-storage to upgrade to wiredTiger./Do not remove'/g /etc/foreman-installer/custom-hiera.yaml")
+  Kafo::Helpers.execute("sed -i -e 's/mmapv1/wiredTiger/g' /etc/foreman-installer/custom-hiera.yaml")
```

Comment:
Do we have to update? Can we just remove this from custom-hiera at this point?

Reply:
Right now we dont have any option set so new installs are getting wiredtiger but since we are not setting the value anywhere my fear is that it wont change to wiredtiger during the installer run, I am just doing this as a safety measure.